src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -199,30 +199,55 @@ private void UpdateSize()
                 if (extraField.Size < sizeof(long))
                     return true;
 
-                long value64 = reader.ReadInt64();
+                // Advancing the stream (by reading from it) is possible only when:
+                // 1. There is an explicit ask to do that (valid files, corresponding boolean flag(s) set to true).
+                // 2. When the size indicates that all the information is available ("slightly invalid files").
+                bool readAllFields = extraField.Size >= sizeof(long) + sizeof(long) + sizeof(long) + sizeof(int);
+
                 if (readUncompressedSize)
-                    zip64Block._uncompressedSize = value64;
+                {
+                    zip64Block._uncompressedSize = reader.ReadInt64();
+                }
+                else if (readAllFields)
+                {
+                    _ = reader.ReadInt64();
```

Comment:
For my own education: is there a perf improvement when using `_ = ReadInt64();` to discard the value, compared to just not using `_`? Or is the idea here to explicitly signal that we are ignoring the value on purpose?

Reply:
It shouldn't have a runtime effect, it's self documenting, both for humans and also for linters/analyzers that might otherwise flag an unused method result.